File: 0242-valid-anagram/0242-valid-anagram.c
bool isAnagram(char* s, char* t) {
    int seen[26] = {0};
    for(int i = 0; i < strlen(s); i++) seen[s[i] - 97]++;
    for(int i = 0; i < strlen(t); i++) seen[t[i] - 97]--;
    for(int i = 0; i < 26; i++)
        if(seen[i] != 0) return false;
    return true; 
}

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character, Integer> map = new HashMap<>();
        for(char ch : s.toCharArray()){
            map.put(ch,map.getOrDefault(ch,0)+1);
        }
        for(char ch : t.toCharArray()){
            if(!map.containsKey(ch)) return false;
            map.put(ch,map.get(ch) -1);
            if(map.get(ch) == 0){
                map.remove(ch);
            }
        }
        return map.isEmpty();
    }
}